pandas/tests/io/test_pytables.py

```diff
@@ -1223,6 +1223,16 @@ def test_append_all_nans(self):
             reloaded = read_hdf(path, 'df_with_missing')
             tm.assert_frame_equal(df_with_missing, reloaded)
 
+    def test_read_missing_key_close_store(self):
+        # GH 25766
+        with ensure_clean_path(self.path) as path:
+            df = pd.DataFrame({'a': range(2), 'b': range(2)})
+            df.to_hdf(path, 'k1')
+
+            pytest.raises(KeyError, pd.read_hdf, path, 'k2')
+
```

Comment:
add a comment that this is smoke test that the file is close before written here

Reply:
updated https://github.com/pandas-dev/pandas/pull/25863/commits/a53b0396504a4c448a5c779dfb5a871cab4ae080